src/Interpreters/MySQL/InterpretersMySQLDDLQuery.cpp

```diff
@@ -117,6 +117,37 @@ static inline NamesAndTypesList getColumnsList(ASTExpressionList * columns_defin
     return columns_name_and_type;
 }
 
+static ColumnsDescription getColumnsDescription(const NamesAndTypesList & columns_name_and_type, const ASTExpressionList * columns_definition)
+{
+    if (columns_name_and_type.size() != columns_definition->children.size())
+            throw Exception("Columns of different size provided.", ErrorCodes::BAD_ARGUMENTS);
+
+    ColumnsDescription columns_description;
+    ColumnDescription column_description;
+    
+    for (
+        auto [column_name_and_type, declare_column_ast] = std::tuple{columns_name_and_type.begin(), columns_definition->children.begin()};
```

Comment:
Oh, I didn't realize that columns_name_and_type was a list. You could consider to just initialize an integer instead of declare_column_ast, and then just use `column_definition->children[i]` to access it.  That would be a little less verbose, but if you prefer this way this is also OK.

Reply:
If it's fine now - I would prefer it this way. Since `NamesAndTypesList` is actual list, so it doesn't support random access, and iterating over index AND iterator at same time looks weird for me.